### ISSUE TITLE
Installation fails on Linux with `undefined reference` linker errors

### ISSUE THREAD

**Issue by norabelrose** [rpy2/rpy2#1069]
**Describe the issue or bug**

I have R installed inside a Miniconda environment. When I try to install from source, I get the following string of error messages.

```
python setup.py install

/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1: undefined reference to `__libc_dlopen_mode@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `__libc_current_sigrtmin_private@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `__libc_dlclose@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1: undefined reference to `__vdso_clock_gettime@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `h_errno@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2: undefined reference to `_dl_sym@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `__libc_longjmp@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `__libc_dl_error_tsd@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `__libc_allocate_rtsig_private@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `__libc_thread_freeres@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/librt.so.1: undefined reference to `__libc_dlsym@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `__libc_current_sigrtmax_private@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2: undefined reference to `_dl_addr@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libdl.so.2: undefined reference to `_dl_vsym@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `__libc_pthread_init@GLIBC_PRIVATE'
/home/nora/miniconda3/compiler_compat/ld: /home/nora/miniconda3/bin/../x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0: undefined reference to `_dl_make_stack_executable@GLIBC_PRIVATE'
collect2: error: ld returned 1 exit status
Traceback (most recent call last):
  File "/home/nora/miniconda3/lib/python3.11/site-packages/setuptools/_distutils/unixccompiler.py", line 266, in link
    self.spawn(linker + ld_args)
  File "/home/nora/miniconda3/lib/python3.11/site-packages/setuptools/_distutils/ccompiler.py", line 1041, in spawn
    spawn(cmd, dry_run=self.dry_run, **kwargs)
  File "/home/nora/miniconda3/lib/python3.11/site-packages/setuptools/_distutils/spawn.py", line 70, in spawn
    raise DistutilsExecError(
distutils.errors.DistutilsExecError: command '/usr/bin/gcc' failed with exit code 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/nora/Code/rpy2/setup.py", line 137, in <module>
    c_extension_status = get_r_c_extension_status(r_home)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nora/Code/rpy2/setup.py", line 117, in get_r_c_extension_status
    status = get_c_extension_status(libraries=c_ext.libraries,
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/nora/Code/rpy2/setup.py", line 82, in get_c_extension_status
    compiler.link_executable(
  File "/home/nora/miniconda3/lib/python3.11/site-packages/setuptools/_distutils/ccompiler.py", line 781, in link_executable
    self.link(
  File "/home/nora/miniconda3/lib/python3.11/site-packages/setuptools/_distutils/unixccompiler.py", line 268, in link
    raise LinkError(msg)
distutils.errors.LinkError: command '/usr/bin/gcc' failed with exit code 1
```

**To Reproduce**

Install `r` and `r-essentials` from `conda-forge` in a Miniconda environment on Manjaro Linux, then try to install this library

**Comment by lgautier** [comment]
This has likely little to do with rpy2, but is caused an incompatibility between system C libraries, miniconda, and possibly dynamic C library dependencies for you conda package providing R. The conda/miniconda, or r/r-essentially discussions channels will be better able to help.

Reopen if it turns out to be an issue with rpy2.

**Comment by function2-llx** [comment]
Okay so I have encountered almost the same issue and I am also using Arch Linux. Here are some hints I've found. 

There are two options to solve this:
1. Set the environment variable `CC=gcc`
2. Install the `gcc` package to the conda environment. 

So what's going on? During setup for `rpy2`, some testing code is compiled and linked:

https://github.com/rpy2/rpy2/blob/58903cf9089c3caf4dee326e1048dc13b20025f5/rpy2-rinterface/setup.py#L68-L87

Here, if `CC` is not set, the default value will become something like `gcc -pthread -B $CONDA_PREFIX/compiler_compat`, used by `setuptools` ([ref code](https://github.com/pypa/setuptools/blob/cc43212ffcc2b3e9830fe4f95d465a17f9f04116/setuptools/_distutils/sysconfig.py#L302-L322)), and which is further managed by [`sysconfig`](https://docs.python.org/3/library/sysconfig.html). As a result, the linker used will be the one provided by the conda environment, from the [`gcc_impl_linux-64`](https://anaconda.org/conda-forge/gcc_impl_linux-64) package (required by `r-base`) or its dependencies. If the compiler used is the one installed on your system, then it will probably cause the inconsistency. 

I don't have enough time to investigate the full details, hope these will at least make things work.
